scripts/metric.py

```diff
@@ -6,63 +6,131 @@
 matrix from CSV output files.
 
 The main function `calc` reads prediction results from a CSV file, generates a confusion matrix
-with comprehensive classification statistics, and outputs all metrics to a CSV file for further analysis.
-
-Example output CSV:
-    | Metric       | Value      |
-    |--------------|------------|
-    | Overall ACC  | 0.85       |
-    | PPV Macro    | 'None'     |
-    | Kappa 95% CI | (0.0, 0.0) |
+with comprehensive classification statistics, and outputs all metrics to both a JSON file for
+further analysis and a plot-ready JSON file for visualization.
+
+Example metric JSON output:
+    {
+        "Overall ACC": "0.85",
+        "PPV Macro": "None",
+        "Kappa 95% CI": "(0.0, 0.0)",
+        ...
+    }
+
+Example plot JSON output:
+    [
+        {"metric": "Overall ACC", "value": 0.85},
+        {"metric": "F1 Macro", "value": "None"},
+        {"metric": "Kappa 95% CI", "value": "(0.0, 0.0)"},
+        ...
+    ]
 
 Functions:
-    calc: Calculate and save performance metrics from prediction output files
+    calculate_all_metrics: Calculate all metrics including confusion matrix and custom metrics
+    calc: Calculate and save performance metrics from prediction output files to metric and plot JSON formats
 """
 
 import argparse
+import json
 from pathlib import Path
 
 import polars as pl
 from pycm import ConfusionMatrix
+from scipy.stats import spearmanr
+
+
+def calculate_all_metrics(
+    actual_values: list,
+    predicted_values: list,
+) -> dict:
+    """Calculate all metrics including confusion matrix metrics and custom metrics.
+
+    This function computes both standard classification metrics via confusion matrix
+    and custom metrics like Spearman correlation. Add new custom metrics here.
+
+    Args:
+        actual_values: List of actual/ground truth values
+        predicted_values: List of predicted values
+
+    Returns:
+        Dictionary of metric names and their values
+    """
+    # Calculate confusion matrix metrics
+    cm = ConfusionMatrix(
```

Comment:
@hredestig : Should we keep this? It was nice to have something, but maybe it is not informative and it clutters the PR with (potentially) irrelevant metrics